protocol/app/prepare/transactions.go

```diff
@@ -33,15 +33,14 @@ func NewPrepareProposalTxs(
 	}
 
 	return PrepareProposalTxs{
-		MaxBytes:  req.MaxTxBytes,
-		UsedBytes: 0,
+		MaxBytes: uint64(req.MaxTxBytes),
```

Comment:
The main reason I ended up using int64 instead of uint64 is because the `PrepareProposalRequest.MaxTxBytes` ([code](https://github.com/cosmos/cosmos-sdk/blob/bff3104781cca3c409fecfa3efebd80a40eb48c3/proto/tendermint/abci/types.proto#L128)) uses `int64` so decided to keep the same data type.

I agree that negative numbers for these do not make sense.

I wonder for the sake of consistency, it's better to keep it as int64 instead of uint64.

@BrendanChou wdyt?

Reply:
We validate that MaxTxBytes can't be negative though.

Reply:
If the `PreparePropoalRequest` is from the ABCI call and is positive, then we proceed with our own internal logic to make sure that we will not return more than the number of bytes requested.

We are not really storing this information anywhere or comparing it to other `int64` types so I think it's fine to convert the type.

Since it is a positive `int64` it will fit into a `uint64` so I see no issue there either.